From 5d5be5a81c3cb3b0bcb80ab086fe37e19a6b20e9 Mon Sep 17 00:00:00 2001
From: Isaac True <isaac.true@nateurope.com>
Date: Fri, 15 Nov 2019 16:39:39 +0100
Subject: [PATCH 0681/1851] drm: xlnx: sdi: Use the named IRQ when available

commit f3eaac34ef1fda1c8648edb167f8c8f6714cec47 from
https://github.com/Xilinx/linux-xlnx.git

This ensures that the correct interrupt is configured when more than one
interrupt is defined in the device tree, such as in the case when both
the vtc_irq and sdi_tx_irq signals are connected in the hardware design.

Signed-off-by: Isaac True <isaac.true@nateurope.com>
Reviewed-by: Vishal Sagar <vishal.sagar@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_sdi.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_sdi.c b/drivers/gpu/drm/xlnx/xlnx_sdi.c
index e97cb43abe44..e9fecb3f2bad 100644
--- a/drivers/gpu/drm/xlnx/xlnx_sdi.c
+++ b/drivers/gpu/drm/xlnx/xlnx_sdi.c
@@ -1144,10 +1144,17 @@ static int xlnx_sdi_probe(struct platform_device *pdev)
 
 	/* disable interrupt */
 	xlnx_sdi_writel(sdi->base, XSDI_TX_GLBL_IER, 0);
-	irq = platform_get_irq(pdev, 0);
+	irq = platform_get_irq_byname(pdev, "sdi_tx_irq");
 	if (irq < 0) {
-		ret = irq;
-		goto err_disable_vidin_clk;
+		/*
+		 * If there is no IRQ with this name, try to get the first
+		 * IRQ defined in the device tree.
+		 */
+		irq = platform_get_irq(pdev, 0);
+		if (irq < 0) {
+			ret = irq;
+			goto err_disable_vidin_clk;
+		}
 	}
 
 	ret = devm_request_threaded_irq(sdi->dev, irq, NULL,
-- 
2.31.1

